src/main/java/org/assertj/core/api/AbstractZonedDateTimeAssert.java

```diff
@@ -459,10 +464,14 @@ public SELF isEqualTo(String dateTimeAsString) {
    *
    * @param expected the given value to compare the actual value to.
    * @return {@code this} assertion object.
+   * @throws AssertionError if the actual {@code ZonedDateTime} is {@code null}.
+   * @throws IllegalArgumentException if other {@code ZonedDateTime} is {@code null}.
    * @throws AssertionError if the actual {@code ZonedDateTime} is equal to the {@link ZonedDateTime} in the actual
    *           ZonedDateTime's java.time.ZoneId.
    */
   public SELF isNotEqualTo(ZonedDateTime expected) {
+    Objects.instance().assertNotNull(info, actual);
```

Comment:
This should be consistent with `AbstractAssert.isNotEqualTo`, if actual is null and expected is not, the assertion should succeed and vice versa.

Reply:
Yes, same as above comment.